### ISSUE TITLE
[UWP] TestPlatform error running on VM in Azure.

### ISSUE THREAD

**Issue by JochemVH1** [Microsoft/vstest#1500]
## Description
Hello, I am currently struggling with running tests for UWP in a build definition in VSTS.
I downloaded the agent and I am running one locally and one in a VM (Windows Server 2016 with Visual Studio) on azure.
The weird part is that when I run the definition on the agent locally everything is fine, and when I run on the agent in the VM I get an error. 

DEP0700: Registration of the app failed. [0x80073CF6] error 0x80070005: Adding a tile failed with unexpected error.

![image](https://user-images.githubusercontent.com/17174475/37839956-1e6c0a1e-2ebc-11e8-8cb8-3339149ff900.png)
![image](https://user-images.githubusercontent.com/17174475/37839976-306336c0-2ebc-11e8-8828-a8c4ad25b9b7.png)
![image](https://user-images.githubusercontent.com/17174475/37839997-3d9c2824-2ebc-11e8-860e-8fb4178da302.png)

Here are the images of the build definition and all the relevant fields I filled in VS Test.

## Steps to reproduce
The solution is currently build up out of the standard UWP test project and a UWP project.
![image](https://user-images.githubusercontent.com/17174475/37840120-86dae5d4-2ebc-11e8-8f13-f5f8fee012f3.png)

I created a Standard B2ms VM (Windows Server 2016 with Visual Studio).
Downloaden the agent from VSTS and configured it on the VM and set it to run as a service.
I am running the services as a local user, otherwise I received a DEP0600 (or DEP0800).

When I try to run the build definition on the agent configured on the VM it somehow fails.

## Expected behavior
![image](https://user-images.githubusercontent.com/17174475/37840385-33455e8a-2ebd-11e8-8eba-973f62e7ff58.png)

This is the behavior I get on the agent running on my local machine.

## Actual behavior
![image](https://user-images.githubusercontent.com/17174475/37840445-61b049c4-2ebd-11e8-9125-a7c9dbadbd9f.png)

This is the behavior I get on the agent running on the VM.

## Diagnostic logs

I am pretty sure I am missing some information. If so, I will provide.

## Environment
![image](https://user-images.githubusercontent.com/17174475/37840659-fa3fa360-2ebd-11e8-9465-8a7f28f5357c.png)

These are the references on the UWP test project.

**Comment by JochemVH1** [comment]
I am pretty sure I am missing something obvious. But I have been at it for 2 days and can't seem to figure it out. Any help is welcome!

**Comment by smadala** [comment]
@JochemVH1 Can you share [build logs](https://github.com/Microsoft/vstest/issues/1480#issuecomment-375404214)? 
What is the  Azure VM VS version? Looks like it is not VS 15.6.

**Comment by JochemVH1** [comment]
@smadala I can confirm that Azure's VM VS version is 15.6.4

![image](https://user-images.githubusercontent.com/17174475/37874475-44ac1a82-3030-11e8-9159-2c836ab9906d.png)

Quick question what is the best way for me to provide you the logs?

**Comment by smadala** [comment]
- Go to failed build summary. 

- Queue a new build with diagnostics logs.
![image](https://user-images.githubusercontent.com/4337332/37877375-e6cff896-3077-11e8-8ca7-e15695266a16.png)


- After build complete, Go to build summary,  click on `Download all logs as zip`.
![image](https://user-images.githubusercontent.com/4337332/37877334-4b1f261a-3077-11e8-85c7-4dea8c39a964.png)

- Share the zip file.

**Comment by JochemVH1** [comment]
[JochemVanHespen_logs_244.zip](https://github.com/Microsoft/vstest/files/1845540/JochemVanHespen_logs_244.zip)
These are the logs from the VM

[JochemVanHespen_logs_245.zip](https://github.com/Microsoft/vstest/files/1845542/JochemVanHespen_logs_245.zip)
These are the logs from the local machine

**Comment by smadala** [comment]
@JochemVH1 Looks like vstest.console failed to launch UWP app on Auzre VM. There may be some settings required to launch UWP app(like Enable Developer mode).

Can you run blank UWP app with same Target version and Min version in Azure VM?

`DEP0700: Registration of the app failed. [0x80073CF6] error 0x80070005: Adding a tile failed with unexpected error.` Can you figure out when this error occurs?

BTW, No need of `VsTest platform tools installer` task for running UWP app and `Microsoft.TestPlatform` nuget reference in you UWP unit test app.

**Comment by JochemVH1** [comment]
Developer mode is enabled. 

I cloned the master branch on to the VM and both the blank UWP app and UWP.Test app are running fine. Also removed the VsTest platform tools installer and nuget reference to Microsoft.TestPlatform.

It is hard to figure out when the DEP0700 is happening because it can not be reproduced through vstest.console.exe on the VM.

**Comment by smadala** [comment]
> It is hard to figure out when the DEP0700 is happening because it can not be reproduced through vstest.console.exe on the VM.

This makes me thinks, Issue is related to how the agent is running as [interactive/service](https://docs.microsoft.com/en-us/vsts/build-release/concepts/agents/agents?view=vsts#account). Azure VM agent running as service?

**Comment by JochemVH1** [comment]
Yes the agent is running as a service. I will try to disable the service and run the agent in powershell.

**Comment by JochemVH1** [comment]
Ow great, indeed it is! Thank you very much. This can be closed.

**Comment by fman028** [comment]
I'm running into the same issue.  I'm not sure what is meant by "run the agent in powershell"?

**Comment by JochemVH1** [comment]
By accident I was just checking some of my closed issues. At that time I had the agent running as a windows service on the virtual machine, but if i manually ran the agent in powershell (so opening powershell and starting the agent) I didnt receive the DEP0700

**Comment by fman028** [comment]
Whoa, I did not expect such a fast response!  Hm, I also have my VSTS Agent running as a service on my virtual machine.  I'm still being a bit dense about stharting the agent from powershell :/  You mean opening poweshell on my vm and running `Start-Service <my agent name>`?  Or having a powershell task in my build pipeline?

**Comment by JochemVH1** [comment]
Yeah it's been a while since ive dealt with this but as far as I remember I ran powershell on the VM and started the agent from there which resolved the issue.

**Comment by fman028** [comment]
Hm, weird still getting the same issue after starting my agent from powershell via `Start-Service <my agent name>` :/  Meh, I'll keep looking.  Appreciate the help on an old post :)
